src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -40,10 +32,43 @@ public static void Run(ConfigPathsArgs args, Func<ILogger> getLogger)
         }
     }
 
+    internal static class ConfigSetRunner
+    {
+        public static void Run(ConfigSetArgs args, Func<ILogger> getLogger)
+        {
+            RunnerHelper.ValidateArguments(args, getLogger);
+            RunnerHelper.ValidateConfigKey(args.ConfigKey);
+            ISettings settings = null;
+            if (string.IsNullOrEmpty(args.ConfigFile))
+            {
+                settings = RunnerHelper.GetSettingsFromDirectory(null);
+            }
+            else
+            {
+                settings = RunnerHelper.GetSettingsFromFile(args.ConfigFile);
+            }
+            var encrypt = false;
+            if (args.ConfigKey.Equals(ConfigurationConstants.PasswordKey, StringComparison.OrdinalIgnoreCase))
+            {
+                encrypt = true;
+            }
+            SettingsUtility.SetConfigValue(settings, args.ConfigKey, args.ConfigValue, encrypt);
+        }
+    }
+
     internal static class RunnerHelper
     {
+        /// <summary>
+        /// Creates a settings object using the directory argument,
+        /// or using the current directory if no argument is passed.
+        /// </summary>
+        /// <exception cref="CommandException"></exception>
         public static ISettings GetSettingsFromDirectory(string directory)
         {
+            if (string.IsNullOrEmpty(directory))
+            {
+                directory = Directory.GetCurrentDirectory();
+            }
```

Comment:
nit: Add a line break
```suggestion
            }
            
```